packages/loader/container-loader/src/container.ts

```diff
@@ -921,6 +924,12 @@ export class Container
 		this.verifyClosed();
 	}
 
+	private switchFromEstablishingConnectionToDisconnect(reason: string) {
```

Comment:
I'm not sure I see the reason for this API and calls to it to be needed.
When we close container (for example), DeltaManager will close connection that will result in all the right events being fired.

Reply:
I think today, ConnectonManager makes decision if we will try to reconnect (on error / disconnect) or not. It definitely does so in cases of container close. While today we can make it work the way you have it in this PR, imagine situations in the future where ConnectionManager will make a call not to reconnect, but also not to close container. In such cases we would need these calls / notifications to come from ConnectionManager.

Similar, if we get 429, there is no right or wrong to attribute time we wait (retryAfter) to being counted as "establishing connection" or not. But I think we have to have a way to implement it either way, and that also tells us to push responsibility to chose that (and notify observers) from ConnectionManager.